### PR TITLE
[Assets] Add normalize/denormalize method for metadata

### LINKED ISSUE #8489: [Assets] Add normalize/denormalize method for metadata
analog to #8487

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #8489

### CODE REVIEW

File: models/Asset/MetaData/ClassDefinition/Data/Asset.php [hunk 1]
```diff
@@ -20,41 +20,57 @@
...
+    }
+
+
     /**
      * @param mixed $value
      * @param array $params
      *
      * @return null|int
+     *
+     * @deprecated use normalize() instead, will be removed in Pimcore 11
```
Comment (blankse) [inline models/Asset/MetaData/ClassDefinition/Data/Asset.php]: We should add trigger_deprecation() in method body

Comment (dvesh3) [reply models/Asset/MetaData/ClassDefinition/Data/Asset.php]: Good catch! done 😉

File: models/Asset/Service.php [hunk 2]
```diff
@@ -22,6 +22,7 @@
 use Pimcore\Model\Asset;
 use Pimcore\Model\Asset\MetaData\ClassDefinition\Data\Data;
 use Pimcore\Model\Element;
+use Pimcore\Normalizer\NormalizerInterface;
```
Comment (fashxp) [inline models/Asset/Service.php]: why do we need that import here?

**Review by fashxp — CHANGES_REQUESTED** [review]
the data retrieved via getter and after a normalize-denormalize cascade should be the same. 

is not the case for metadata of type `asset`, `document` and `object`

we should also add a test for that  ... snippet for a headstart of that test could be as follows: 

```php 
        $loader = \Pimcore::getContainer()->get('pimcore.implementation_loader.asset.metadata.data');
        
        $asset = Asset::getById(68);
        $metadataList = $asset->getMetadata();

        foreach($metadataList as $metaData) {
            try {
                /** @var Data $instance */
                $instance = $loader->build($metaData['type']);

                $marshalledData = $instance->normalize($metaData['data']);
                $unmarshalledData = $instance->denormalize($marshalledData);

                p_r($metaData['name'] . ': ');
                p_r($unmarshalledData === $metaData['data']);
                p_r("-------------------");

            } catch (UnsupportedException $e) {
            }


        }
```

File: models/Asset/Service.php (hunk 2 above)
Comment (dvesh3) [reply models/Asset/Service.php]: yes, we can get rid of it,

**Review by fashxp — APPROVED** [review]

**Pull request merged**